src/Sylius/Component/User/Security/TokenProvider.php

```diff
@@ -73,6 +73,6 @@ public function generateUniqueToken()
      */
     protected function isUsedCode($token)
     {
-        return null !== $this->repository->findOneBy(['confirmationToken' => $token]);
+        return null !== $this->repository->findOneByToken($token);
```

Comment:
This `TokenProvider` should have two methods, `generateUniqueEmailVerificationToken` and `generateUniquePasswordResetToken`.